### PR TITLE
[Fabric] APIs to enable RootViews that size to content

### PR DESCRIPTION
**Pull request by acoates-ms**
Modifies the APIs on CompositionRootView to enable hosting RN content where the container can size based on the RN content, rather than forcing the RN content to the size of the container.

The Measure/Arrange APIs on CompositionRootView align with the APIs on SurfaceHandler (https://github.com/facebook/react-native/blob/b7fc5867f2f8248db8230ce3602077547ca971a1/packages/react-native/ReactCommon/react/renderer/scheduler/SurfaceHandler.h#L131) which they eventually call into.

An additional event for SizeChanged is added to CompositionRootView to enable the container to listen for when the content's size changes, and resize the container.

I added a new option in Playground's settings to have the playground app size itself based on the RN content rather than the normal method of having the RN content fit into the window.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/react-native-windows/pull/13269)

### CODE REVIEW

File: vnext/Microsoft.ReactNative/Fabric/Composition/CompositionRootView.cpp [hunk 1]
```diff
@@ -493,19 +474,82 @@ void CompositionRootView::ShowInstanceLoaded() noexcept {
...
+}
+
+facebook::react::Size MeasureLoading(const facebook::react::LayoutConstraints &layoutConstraints, float scaleFactor) {
+  auto attributedStringBox = CreateLoadingAttributedString();
+  winrt::com_ptr<::IDWriteTextLayout> textLayout;
+  facebook::react::TextLayoutManager::GetTextLayout(
+      attributedStringBox, {} /*paragraphAttributes*/, layoutConstraints, textLayout);
+
+  DWRITE_TEXT_METRICS tm;
+  textLayout->GetMetrics(&tm);
```
Comment (marlenecota) [inline vnext/Microsoft.ReactNative/Fabric/Composition/CompositionRootView.cpp:510]: check_hresult?

File: vnext/Microsoft.ReactNative/Fabric/Composition/CompositionRootView.cpp [hunk 2]
```diff
@@ -493,19 +474,82 @@ void CompositionRootView::ShowInstanceLoaded() noexcept {
...
+        &handler) noexcept {
+  return m_sizeChangedEvent.add(handler);
+}
+
+void CompositionRootView::SizeChanged(winrt::event_token const &token) noexcept {
+  m_sizeChangedEvent.remove(token);
+}
+
+struct RootViewSizeChangedEventArgs : RootViewSizeChangedEventArgsT<RootViewSizeChangedEventArgs> {
+  RootViewSizeChangedEventArgs(winrt::Windows::Foundation::Size size) : m_size(size) {}
```
Comment (marlenecota) [inline vnext/Microsoft.ReactNative/Fabric/Composition/CompositionRootView.cpp:527]:
```suggestion
  RootViewSizeChangedEventArgs(const winrt::Windows::Foundation::Size &size) : m_size(size) {}
```

File: packages/e2e-test-app-fabric/windows/RNTesterApp-Fabric/RNTesterApp-Fabric.cpp [hunk 3]
```diff
@@ -50,8 +50,9 @@ void UpdateRootViewSizeToAppWindow(
   // Do not relayout when minimized
   if (window.Presenter().as<winrt::Microsoft::UI::Windowing::OverlappedPresenter>().State() !=
       winrt::Microsoft::UI::Windowing::OverlappedPresenterState::Minimized) {
-    rootView.Arrange(size);
-    rootView.Size(size);
+    winrt::Microsoft::ReactNative::LayoutConstraints constraints;
+    constraints.MaximumSize = constraints.MinimumSize = size;
+    rootView.Arrange(constraints, {0, 0});
```
Comment (marlenecota) [inline packages/e2e-test-app-fabric/windows/RNTesterApp-Fabric/RNTesterApp-Fabric.cpp:55]: do we expect the viewportoffset will always be {0,0}?

Comment (acoates-ms) [reply packages/e2e-test-app-fabric/windows/RNTesterApp-Fabric/RNTesterApp-Fabric.cpp:55]: No.  In cases where the rootview is not taking up the entire content of the host, or in particular is not positioned at 0,0 within the host, then the viewportOffset would be non-zero.

**Review by marlenecota — APPROVED** [review]

**Pull request merged**